script/tool/lib/src/drive_examples_command.dart

```diff
@@ -43,213 +48,252 @@ class DriveExamplesCommand extends PluginCommand {
+            .cast<Map<String, dynamic>>();
+    for (final Map<String, dynamic> deviceInfo in devices) {
+      final String targetPlatform =
+          (deviceInfo['targetPlatform'] as String?) ?? '';
+      if (targetPlatform.startsWith(platform)) {
+        final String? deviceId = deviceInfo['id'] as String?;
+        if (deviceId != null) {
+          deviceIds.add(deviceId);
+        }
       }
-      throw ToolExit(1);
     }
+    return deviceIds;
+  }
+
+  Future<List<File>> _getDrivers(Directory example) async {
+    final List<File> drivers = <File>[];
 
-    if (pluginsWithoutTests.isNotEmpty) {
-      print('The following plugins did not run any integration tests:');
-      for (final String plugin in pluginsWithoutTests) {
-        print(' * $plugin');
+    final Directory driverDir = example.childDirectory('test_driver');
+    if (driverDir.existsSync()) {
+      await for (final FileSystemEntity driver in driverDir.list()) {
+        if (driver is File && driver.basename.endsWith('_test.dart')) {
+          drivers.add(driver);
+        }
       }
-      print('If this is intentional, they must be explicitly excluded.');
-      throw ToolExit(1);
     }
+    return drivers;
+  }
+
+  File? _getLegacyTestFileForTestDriver(File testDriver) {
+    final String testName = testDriver.basename.replaceAll(
+      RegExp(r'_test.dart$'),
+      '.dart',
+    );
+    final File testFile = testDriver.parent.childFile(testName);
 
-    print('All driver tests successful!');
+    return testFile.existsSync() ? testFile : null;
   }
 
-  Future<bool> _pluginSupportedOnCurrentPlatform(
-      FileSystemEntity plugin) async {
-    final bool isAndroid = getBoolArg(kPlatformAndroid);
-    final bool isIOS = getBoolArg(kPlatformIos);
-    final bool isLinux = getBoolArg(kPlatformLinux);
-    final bool isMacos = getBoolArg(kPlatformMacos);
-    final bool isWeb = getBoolArg(kPlatformWeb);
-    final bool isWindows = getBoolArg(kPlatformWindows);
-    if (isAndroid) {
-      return isAndroidPlugin(plugin);
-    }
-    if (isIOS) {
-      return isIosPlugin(plugin);
-    }
-    if (isLinux) {
-      return isLinuxPlugin(plugin);
-    }
-    if (isMacos) {
-      return isMacOsPlugin(plugin);
-    }
-    if (isWeb) {
-      return isWebPlugin(plugin);
+  Future<List<File>> _getIntegrationTests(Directory example) async {
+    final List<File> tests = <File>[];
+    final Directory integrationTestDir =
+        example.childDirectory('integration_test');
+
+    if (integrationTestDir.existsSync()) {
+      await for (final FileSystemEntity file in integrationTestDir.list()) {
+        if (file is File && file.basename.endsWith('_test.dart')) {
+          tests.add(file);
+        }
+      }
     }
-    if (isWindows) {
-      return isWindowsPlugin(plugin);
+    return tests;
+  }
+
+  /// Runs each file from [targets] using `flutter drive`, returning a list of
+  /// any failing test targets.
```

Comment:
nit: if you are going to have a docstring, it helps to mention all the arguments.  It's kind of hard to guess what they all mean.

Reply:
Done.